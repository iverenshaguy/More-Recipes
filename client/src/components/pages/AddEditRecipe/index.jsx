import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Form from '../../shared/Forms';
import { urlMatchPropTypes } from '../../../helpers/proptypes';

/**
 * @exports
 * @class AddEditRecipe
 * @extends Component
 * @returns {component} AddEditRecipe
 */
class AddEditRecipe extends Component {
  static propTypes = {
    ...urlMatchPropTypes,
    type: PropTypes.string,
    uploading: PropTypes.bool.isRequired,
    submitting: PropTypes.bool.isRequired,
  };

  static defaultProps = {
    type: 'addRecipe'
  }

  /**
   * @memberof AddEditRecipe
   * @returns {component} AddEditRecipe
   */
  render() {
    const meta = {
      title: this.props.type === 'addRecipe' ? 'Add a New Recipe' : 'Edit Recipe',
      btnText: this.props.type === 'addRecipe' ? 'Add Recipe' : 'Edit Recipe'
    };

    return (
      <div className="card-form-body">
        <div className="container">
          <div className="my-5" id="card-form-wrapper">
            <div className="p-xs-2 pt-4 pb-3" id="login-signup-div">
              <Form {...this.props} type={this.props.type} meta={meta} />
            </div>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  submitting: state.singleRecipe.recipe.adding,
  uploading: state.uploadImage.uploading,
  submitError: state.singleRecipe.recipe.error
});

export { AddEditRecipe as AddEditRecipeComponent };
export default connect(mapStateToProps)(AddEditRecipe);
