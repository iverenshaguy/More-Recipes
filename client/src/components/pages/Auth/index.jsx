import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link, Redirect } from 'react-router-dom';
import { Row, Col } from 'reactstrap';
import Form from '../../shared/Forms';
import setCurrentLocation from '../../../actions/location';
import { authPropTypes } from '../../../helpers/proptypes';
import './Auth.scss';

/**
 * @exports
 * @class Auth
 * @extends Component
 * @returns {component} Auth
 */
class Auth extends Component {
  static propTypes = {
    ...authPropTypes,
    type: PropTypes.string.isRequired,
    dispatch: PropTypes.func.isRequired,
  };

  /**
   * @memberof LoginForm
   * @returns {nothing} Returns nothing
   */
  componentWillMount() {
    this.props.dispatch(setCurrentLocation('auth'));
  }

  /**
   * @memberof Auth
   * @returns {component} Auth
   */
  render() {
    const { from } = this.props.location.state ? this.props.location.state : { from: { pathname: '/' } };

    if (this.props.isAuthenticated) {
      return <Redirect to={from} />;
    }

    const signupMeta = {
      title: 'Register for a New Account',
      btnText: 'SIGN UP',
      extra: <p className="text-center">Already have an account, <Link to="/login">Log in here</Link>.</p>
    };

    const loginMeta = {
      title: 'Sign In to Your Account',
      btnText: 'SIGN IN',
      extra: <p className="text-center">Dont have an account, <Link to="/signup">Sign up here</Link>.</p>
    };

    return (
      <div className="login-signup-body">
        <div className="container" id="login-signup-container">
          <Row className="my-5" id="login-signup-wrapper">
            <Col className="p-xs-2 p-lg-4" id="login-signup-div">
              {this.props.type === 'signup' ?
                <Form {...this.props} type="signup" meta={signupMeta} /> :
                <Form {...this.props} type="login" meta={loginMeta} />}
            </Col>
          </Row>
        </div>
      </div>
    );
  }
}

export { Auth as AuthComponent };

const mapStateToProps = state => ({
  submitting: state.auth.loading,
  isAuthenticated: state.auth.isAuthenticated,
  submitError: state.auth.error
});

export default connect(mapStateToProps)(Auth);
