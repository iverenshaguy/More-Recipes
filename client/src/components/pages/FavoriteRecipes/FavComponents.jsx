import React from 'react';
import { Switch, Route } from 'react-router-dom';
import RecipeItems from '../../shared/RecipeItems';
import { userPropTypes, multiRecipePropTypes } from '../../../helpers/proptypes';

/**
 * @exports
 * @function Fav
 * @param {object} props - props
 * @desc Returns User Favorite Components Route
 * @returns {JSX} User Favorite Components
 */
const FavComponents = props => (
  <Switch>
    <Route
      path={`${props.match.path}/categories`}
      render={() => (<RecipeItems
        title="RECIPE CATEGORIES"
        recipes={props.recipes}
        handlePageChange={props.handlePageChange}
        metadata={props.metadata}
      />)}
    />
    <Route
      path={props.match.path}
      render={() => (<RecipeItems
        title="RECIPES"
        recipes={props.recipes}
        handlePageChange={props.handlePageChange}
        metadata={props.metadata}
      />)}
    />
  </Switch>);

FavComponents.propTypes = {
  ...userPropTypes,
  ...multiRecipePropTypes,
};

export default FavComponents;
