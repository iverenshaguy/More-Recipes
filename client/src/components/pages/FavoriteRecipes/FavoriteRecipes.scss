@import '../../../assets/scss/modules';


.favorite-recipes {
  background-color: $background-grey !important;
}