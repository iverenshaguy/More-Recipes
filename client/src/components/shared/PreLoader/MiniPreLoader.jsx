import React from 'react';
import Loader from 'react-loaders';
import 'loaders.css/src/animations/ball-beat.scss';
import './PreLoader.scss';

const MiniPreLoader = () => (<Loader type="ball-beat" />);

export default MiniPreLoader;
