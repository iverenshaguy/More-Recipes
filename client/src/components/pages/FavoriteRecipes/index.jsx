import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { push } from 'react-router-redux';
import FontAwesome from 'react-fontawesome';
import RecipeItems from '../../shared/RecipeItems';
import { MiniPreLoader } from '../../shared/PreLoader';
import { fetchFavoriteRecipes } from '../../../actions/recipes';
import { userPropTypes, multiRecipePropTypes, urlMatchPropTypes } from '../../../helpers/proptypes';
import './FavoriteRecipes.scss';

/**
 * @exports
 * @class FavoriteRecipes
 * @extends Component
 * @classdesc Returns User FavoriteRecipes Page
 * @returns {JSX} User FavoriteRecipes
 */
class FavoriteRecipes extends Component {
  static propTypes = {
    ...userPropTypes,
    ...urlMatchPropTypes,
    ...multiRecipePropTypes,
    isFetching: PropTypes.bool.isRequired,
    dispatch: PropTypes.func.isRequired
  }

  /**
   * @memberof FavoriteRecipes
   * @constructor
   * @returns {JSX} FavoriteRecipes
   */
  constructor() {
    super();

    this.state = {
      limit: 5,
      currentPage: 1,
      activeTab: 'recipes'
    };

    this.toggleTab = this.toggleTab.bind(this);
    this.handlePageChange = this.handlePageChange.bind(this);
    // this.showModal = this.showModal.bind(this);
  }

  /**
   * @memberof FavoriteRecipes
   * @returns {nothing} returns nothing
   */
  componentWillMount() {
    const { match, user, dispatch } = this.props;

    if ((match.path).includes('categories')) {
      dispatch(push(`/${user.username}/favorites/categories`));

      this.setState({
        activeTab: 'categories'
      });
    } else if (match.params.username !== user.username) {
      // redirect user to real username if a new uer logins after an old user
      dispatch(push(`/${user.username}/favorites`));
    }

    dispatch(fetchFavoriteRecipes(this.props.user.id, this.state.currentPage, this.state.limit));
  }

  /**
   * @memberof Home
   * @param {number} page
   * @returns {state} home
   */
  handlePageChange(page) {
    this.setState({
      currentPage: page
    }, () => this.props.dispatch(fetchFavoriteRecipes(this.props.user.id, page, this.state.limit)));
  }

  // /**
  //  * @memberof FavoriteRecipes
  //  * @param {string} modal - modal to show
  //  * @returns {nothing} Returns nothing
  //  */
  // showModal(modal) {
  //   this.props.dispatch(toggleModal(modal));
  // }

  /**
   * @memberof FavoriteRecipes
   * @param {object} e - event
   * @param {string} activeTab - tab
   * @returns {nothing} Returns nothing
   */
  toggleTab(e, activeTab) {
    e.preventDefault();
    const { user } = this.props;
    const location = activeTab === 'recipes' ? `/${user.username}/favorites` : `/${user.username}/favorites/categories`;

    this.setState({ activeTab });
    this.props.dispatch(push(location));
  }

  /**
   * @memberof FavoriteRecipes
   * @returns {JSX} User FavoriteRecipes
   */
  render() {
    const { activeTab } = this.state;

    const {
      isFetching, recipes, metadata, user
    } = this.props;

    const recipesTab = classNames({
      'd-inline': true,
      'recipes-tab': true,
      'text-center': true,
      active: activeTab === 'recipes'
    });

    const categoriesTab = classNames({
      'd-inline': true,
      'recipes-tab': true,
      'text-center': true,
      active: activeTab === 'categories'
    });

    return (
      <div className="favorite-recipes pb-4">
        <h3 className="text-center pt-5">FAVORITE RECIPES</h3>
        <div className="profile-toggle-wrapper text-center mt-3">
          <Link to={`/${user.username}`} className="back-link"><FontAwesome name="arrow-left" tag="i" /> BACK TO PROFILE</Link>
          <a className={recipesTab} href="#recipes" onClick={e => this.toggleTab(e, 'recipes')}>
            <i className="aria-hidden flaticon flaticon-flat-plate-with-hot-food-from-side-view" />
            <span className="d-none d-sm-inline">&nbsp;&nbsp;RECIPES</span>
          </a>
          <a className={categoriesTab} href="#categories" onClick={e => this.toggleTab(e, 'categories')}>
            <FontAwesome name="folder" size="lg" tag="i" />
            <span className="d-none d-sm-inline">&nbsp;&nbsp;CATEGORIES</span>
          </a>
        </div>
        <div className="container-fluid text-center user-profile-recipe-cards-wrapper mb-5 mt-1">
          {isFetching &&
            <div className="justify-content-center py-5">
              <MiniPreLoader />
            </div>}
          {!isFetching &&
            <div className="row justify-content-center user-profile-recipe-cards px-5 px-sm-5 px-md-0 mt-0">
              {activeTab === 'recipes' && (
                <RecipeItems
                  title="RECIPES"
                  recipes={recipes}
                  handlePageChange={this.handlePageChange}
                  metadata={metadata}
                />
              )}
              {activeTab === 'categories' && (
                <RecipeItems
                  title="RECIPE CATEGORIES"
                  recipes={recipes}
                  handlePageChange={this.handlePageChange}
                  metadata={metadata}
                />
              )}
            </div>}
        </div>
      </div>);
  }
}

const mapStateToProps = state => ({
  user: state.auth.user,
  isFetching: state.isFetching,
  recipes: state.recipes.items,
  metadata: state.recipes.metadata,
});

export { FavoriteRecipes as FavoriteRecipesComponent };
export default connect(mapStateToProps)(FavoriteRecipes);
