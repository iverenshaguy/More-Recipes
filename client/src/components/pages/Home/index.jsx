import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import setCurrentLocation from '../../../actions/location';
import { fetchTopRecipes, searchRecipes } from '../../../actions/recipes';
import Hero from './Hero';
import RecipeItems from '../../shared/RecipeItems';
import PreLoader from '../../shared/PreLoader';
import { multiRecipePropTypes } from '../../../helpers/proptypes';
import './Home.scss';

/**
 * @exports
 * @class Home
 * @extends Component
 * @returns {component} Home
 */
class Home extends Component {
  static propTypes = {
    dispatch: PropTypes.func.isRequired,
    isFetching: PropTypes.bool.isRequired,
    ...multiRecipePropTypes
  };

  /**
   * @constructor
   * @memberof Home
   * @returns {nothing} - Returns nothing
   */
  constructor() {
    super();

    this.state = {
      title: 'TOP RECIPES',
      currentPage: 1,
      limit: 5,
      searchValue: ''
    };

    this.handlePageChange = this.handlePageChange.bind(this);
    this.handleSearchInput = this.handleSearchInput.bind(this);
    this.handleSearch = this.handleSearch.bind(this);
  }

  /**
   * @memberof Home
   * @return {nothing} Returns nothing
   */
  componentWillMount() {
    this.props.dispatch(setCurrentLocation('home'));
    this.props.dispatch(fetchTopRecipes(this.state.currentPage, this.state.limit));
  }

  /**
   * @memberof Home
   * @param {number} page
   * @returns {state} home
   */
  handlePageChange(page) {
    let fetch;
    const { limit, searchValue } = this.state;

    if (this.state.title === 'TOP RECIPES') {
      fetch = fetchTopRecipes(page, limit);
    } else {
      fetch = searchRecipes(searchValue, page, limit);
    }

    this.setState({
      currentPage: page
    }, () => this.props.dispatch(fetch));
  }

  /**
   * @memberof Home
   * @param {event} event
   * @returns {state} home
   */
  handleSearchInput(event) {
    this.setState({
      searchValue: event.target.value
    });
  }

  /**
   * @memberof Home
   * @param {object} event
   * @returns {state} home
   */
  handleSearch(event) {
    event.preventDefault();

    const { searchValue, currentPage, limit } = this.state;
    this.setState({
      title: 'SEARCH RESULTS'
    }, () => this.props.dispatch(searchRecipes(searchValue, currentPage, limit)));
  }

  /**
   * @memberof Home
   * @returns {fragment} Recipe items
   */
  renderBody() {
    const { isFetching, recipes, metaData } = this.props;
    const { title } = this.state;

    if (isFetching) {
      return <PreLoader />;
    }

    return (
      <RecipeItems
        title={title}
        recipes={recipes}
        handlePageChange={this.handlePageChange}
        metaData={metaData}
      />
    );
  }

  /**
   * @memberof Home
   * @returns {component} Home
   */
  render() {
    return (
      <div className="home">
        <Hero
          dispatch={this.props.dispatch}
          handleSearchInput={this.handleSearchInput}
          handleSearch={this.handleSearch}
          searchValue={this.state.searchValue}
        />
        {this.renderBody()}
      </div>
    );
  }
}

const mapStateToProps = state => ({
  isFetching: state.isFetching,
  recipes: state.recipes.recipes,
  metaData: state.recipes.metaData
});

export { Home as HomeComponent };

export default connect(mapStateToProps)(Home);
