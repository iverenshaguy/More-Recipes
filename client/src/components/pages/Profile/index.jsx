import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Link } from 'react-router-dom';
import { push } from 'react-router-redux';
import { Button } from 'reactstrap';
import PropTypes from 'prop-types';
import FontAwesome from 'react-fontawesome';
import ProfilePic from './ProfilePic';
import RecipeItems from '../../shared/RecipeItems';
import { MiniPreLoader } from '../../shared/PreLoader';
import { toggleModal } from '../../../actions/ui';
import { fetchUserRecipes } from '../../../actions/recipes';
import { updateUserImage } from '../../../actions/auth';
import {
  setUploading, unsetUploading, uploadSuccess, uploadFailure, clearUploadError, uploadImage
} from '../../../actions/uploadImage';
import { userPropTypes, multiRecipePropTypes, urlMatchPropTypes } from '../../../helpers/proptypes';
import './Profile.scss';

/**
 * @exports
 * @class Profile
 * @extends Component
 * @classdesc Returns User Profile Page
 * @returns {JSX} User Profile
 */
class Profile extends Component {
  static propTypes = {
    ...userPropTypes,
    ...urlMatchPropTypes,
    ...multiRecipePropTypes,
    isFetching: PropTypes.bool.isRequired,
    updateUserImage: PropTypes.func.isRequired
  }

  /**
   * @memberof Profile
   * @constructor
   * @returns {JSX} Profile
   */
  constructor() {
    super();

    this.state = {
      limit: 5,
      currentPage: 1
    };

    this.handlePageChange = this.handlePageChange.bind(this);
    this.showAddRecipeModal = this.showAddRecipeModal.bind(this);
  }

  /**
   * @memberof Profile
   * @returns {nothing} returns nothing
   */
  componentWillMount() {
    const { match, user, dispatch } = this.props;
    if (match.params.username !== user.username) {
      dispatch(push(`/${user.username}`));
    }

    dispatch(fetchUserRecipes(this.props.user.id, this.state.currentPage, this.state.limit));
  }

  /**
   * @memberof Home
   * @param {number} page
   * @returns {state} home
   */
  handlePageChange(page) {
    this.setState({
      currentPage: page
    }, () => this.props.dispatch(fetchUserRecipes(this.props.user.id, page, this.state.limit)));
  }

  /**
   * @memberof Profile
   * @returns {nothing} Returns nothing
   */
  showAddRecipeModal() {
    this.props.dispatch(toggleModal('addRecipe'));
  }

  /**
   * @memberof Profile
   * @returns {JSX} User Profile
   */
  render() {
    const {
      user, isFetching, recipes, metadata, uploadImageObj
    } = this.props;

    return (
      <div className="user-profile pb-4">
        <div className="container-fluid user-profile-div">
          <div className="row justify-content-start user-info py-4 px-3 px-md-5">
            <ProfilePic
              user={user}
              uploadImageObj={uploadImageObj}
              uploadImage={this.props.uploadImage}
              setUploading={this.props.setUploading}
              uploadFailure={this.props.uploadFailure}
              uploadSuccess={this.props.uploadSuccess}
              unsetUploading={this.props.unsetUploading}
              updateUserImage={this.props.updateUserImage}
              clearUploadError={this.props.clearUploadError}
            />
            <div className="col-7 col-md-8 name-div align-self-center">
              <p className="name pt-2">{`${user.firstname} ${user.lastname}`}</p>
              <small className="text-muted username pt-1">{`@${user.username}`}</small>
              <p className="about-me pt-2">{user.aboutMe}</p>
            </div>
          </div>
        </div>
        <div className="container-fluid text-center user-profile-recipe-cards-wrapper mb-5 mt-1">
          <Button
            onClick={this.showAddRecipeModal}
            className="btn-default btn-lg d-none d-md-inline-block"
            id="home-add-recipe-btn"
            title="New Recipe"
          >
            Add a New Recipe
          </Button>
          <div className="profile-toggle-wrapper text-center mt-3">
            <a className="d-inline recipes-tab text-center active" href="#recipes">
              <i className="aria-hidden flaticon flaticon-flat-plate-with-hot-food-from-side-view" />
              <span className="d-none d-sm-inline">&nbsp;&nbsp;RECIPES</span>
            </a>
            <Link to={`/${user.username}/favorites`} className="d-inline recipes-tab text-center" href="#favorite-recipes">
              <FontAwesome name="heart" size="lg" tag="i" />
              <span className="d-none d-sm-inline">&nbsp;&nbsp;FAVORITES</span>
            </Link>
          </div>
          {isFetching &&
            <div className="justify-content-center py-5">
              <MiniPreLoader />
            </div>}
          {!isFetching &&
            <div className="row justify-content-center user-profile-recipe-cards px-5 px-sm-5 px-md-0 mt-0">
              <RecipeItems
                title="MY RECIPES"
                recipes={recipes}
                handlePageChange={this.handlePageChange}
                metadata={metadata}
              />
            </div>}
        </div>
      </div>);
  }
}

const mapStateToProps = state => ({
  user: state.auth.user,
  isFetching: state.isFetching,
  recipes: state.recipes.items,
  metadata: state.recipes.metadata,
  uploadImageObj: state.uploadImage
});

const mapDispatchToProps = dispatch => bindActionCreators({
  clearUploadError,
  updateUserImage,
  unsetUploading,
  uploadSuccess,
  uploadFailure,
  setUploading,
  uploadImage
}, dispatch);

export { Profile as ProfileComponent };
export default connect(mapStateToProps, mapDispatchToProps)(Profile);
