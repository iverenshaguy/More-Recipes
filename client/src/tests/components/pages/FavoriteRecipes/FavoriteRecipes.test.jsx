import React from 'react';
import MiniPreLoader from '../../../../components/shared/PreLoader/MiniPreLoader';
import FavComponents from '../../../../components/pages/FavoriteRecipes/FavComponents';
import { FavoriteRecipesComponent } from '../../../../components/pages/FavoriteRecipes';

const setup = () => {
  const props = {
    dispatch: jest.fn(),
    match: {
      isExact: true,
      url: '/favorites',
      params: {
        username: 'username'
      }
    },
    isFetching: false,
    recipes: [{ id: 1, recipeName: 'Rice', User: 'Me' }, { id: 5, recipeName: 'Beans', User: 'She' }],
    user: {
      id: 2,
      username: 'username'
    },
  };

  const shallowWrapper = shallow(<FavoriteRecipesComponent {...props} />, rrcMock.get());
  const mountedWrapper = mount(<FavoriteRecipesComponent {...props} />, rrcMock.get());

  return { props, shallowWrapper, mountedWrapper };
};

// const state = {
//   limit: 1,
//   currentPage: 1,
//   activeTab: 'recipes'
// };

describe('FavoriteRecipes', () => {
  it('renders correctly', () => {
    const { shallowWrapper } = setup();
    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });

  it('renders MiniPreloader when isFetching is true', () => {
    const { props } = setup();
    const shallowWrapper = shallow(<FavoriteRecipesComponent {...props} isFetching />);

    expect(shallowWrapper.find(MiniPreLoader).length).toBeTruthy();
    expect(shallowWrapper.find(FavComponents).length).toBeFalsy();
  });

  it('renders FavComponents when isFetching is false and there are recipe', () => {
    const { props } = setup();
    const shallowWrapper = shallow(<FavoriteRecipesComponent
      {...props}
      isFetching={false}
    />);

    expect(shallowWrapper.find(MiniPreLoader).length).toBeFalsy();
    expect(shallowWrapper.find(FavComponents).length).toBeTruthy();
  });

  it('calls component will mount when mounted', () => {
    const { mountedWrapper, props } = setup();
    expect(props.dispatch).toHaveBeenCalled();
    mountedWrapper.unmount();
  });

  it('redirects to page when username is different', () => {
    const { props } = setup();
    const match = {
      isExact: false,
      url: '/favorites',
      params: {
        username: 'username2'
      }
    };

    const mountedWrapper = mount(<FavoriteRecipesComponent
      {...props}
      match={match}
    />, rrcMock.get());

    expect(mountedWrapper.find('a[href="#recipes"]').hasClass('active')).toBeTruthy();
    expect(mountedWrapper.find('a[href="#categories"]').hasClass('active')).toBeFalsy();
    mountedWrapper.unmount();
  });

  it('renders categories when categories is passed to url', () => {
    const { props } = setup();
    const match = {
      isExact: false,
      url: '/favorites',
      params: {
        username: 'username'
      }
    };

    const mountedWrapper = mount(<FavoriteRecipesComponent
      {...props}
      match={match}
    />, rrcMock.get());

    expect(mountedWrapper.find('a[href="#recipes"]').hasClass('active')).toBeFalsy();
    expect(mountedWrapper.find('a[href="#categories"]').hasClass('active')).toBeTruthy();
    mountedWrapper.unmount();
  });

  // it('calls showModal', () => {
  //   const { shallowWrapper, dispatch } = setup();

  //   shallowWrapper.find('#home-add-recipe-btn').simulate('click');
  //   expect(dispatch).toHaveBeenCalledTimes(1);
  //   expect(dispatch).toHaveBeenCalledWith(toggleModal('addRecipe'));
  // });

  it('calls toggleTab when link is clicked', () => {
    const { mountedWrapper } = setup();
    const toggleTabSpy = jest.spyOn(mountedWrapper.instance(), 'toggleTab');

    mountedWrapper.find('a[href="#categories"]').simulate('click');
    expect(toggleTabSpy).toHaveBeenCalled();

    mountedWrapper.find('a[href="#recipes"]').simulate('click');
    expect(toggleTabSpy).toHaveBeenCalled();
  });

  // it('calls voteRecipe() when upvote icon is clicked', () => {
  //   const { mountedWrapper, props } = setup();

  //   mountedWrapper.find('i.fa-thumbs-up').simulate('click');

  //   expect(props.dispatch).toHaveBeenCalled();
  // });

  // it('calls voteRecipe() when downvote icon is clicked', () => {
  //   const { mountedWrapper, props } = setup();

  //   mountedWrapper.find('i.fa-thumbs-down').at(1).simulate('click');

  //   expect(props.dispatch).toHaveBeenCalled();
  // });
});
