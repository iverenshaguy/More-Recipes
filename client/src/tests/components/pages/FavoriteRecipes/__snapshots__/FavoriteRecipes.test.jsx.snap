// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteRecipes renders correctly 1`] = `
<div
  className="favorite-recipes pb-4"
>
  <h3
    className="text-center pt-5"
  >
    FAVORITE RECIPES
  </h3>
  <div
    className="profile-toggle-wrapper text-center mt-3"
  >
    <Link
      className="back-link"
      replace={false}
      to="/username"
    >
      <FontAwesome
        name="arrow-left"
        tag="i"
      />
       BACK TO PROFILE
    </Link>
    <a
      className="d-inline recipes-tab text-center active"
      href="#recipes"
      onClick={[Function]}
    >
      <i
        className="aria-hidden flaticon flaticon-flat-plate-with-hot-food-from-side-view"
      />
      <span
        className="d-none d-sm-inline"
      >
          RECIPES
      </span>
    </a>
    <a
      className="d-inline recipes-tab text-center"
      href="#categories"
      onClick={[Function]}
    >
      <FontAwesome
        name="folder"
        size="lg"
        tag="i"
      />
      <span
        className="d-none d-sm-inline"
      >
          CATEGORIES
      </span>
    </a>
  </div>
  <div
    className="container-fluid text-center user-profile-recipe-cards-wrapper mb-5 mt-1"
  >
    <div
      className="row justify-content-center user-profile-recipe-cards px-5 px-sm-5 px-md-0 mt-0"
    >
      <FavComponents
        dispatch={[Function]}
        handlePageChange={[Function]}
        isFetching={false}
        match={
          Object {
            "isExact": true,
            "params": Object {
              "username": "username",
            },
            "url": "/favorites",
          }
        }
        recipes={
          Array [
            Object {
              "User": "Me",
              "id": 1,
              "recipeName": "Rice",
            },
            Object {
              "User": "She",
              "id": 5,
              "recipeName": "Beans",
            },
          ]
        }
        user={
          Object {
            "id": 2,
            "username": "username",
          }
        }
      />
    </div>
  </div>
</div>
`;
